tests/phpunit/tests/plugins/test-translated-table.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+class Translated_Table_Test extends PLL_UnitTestCase {
+
+	public static function pllSetUpBeforeClass( PLL_UnitTest_Factory $factory ) {
+		parent::pllSetUpBeforeClass( $factory );
+		$factory->language->create_many( 2 );
+
+		// Translatable custom table.
+		require_once PLL_TEST_DATA_DIR . 'translatable.php';
+	}
+
+	/**
+	 * @ticket #2713 {@see https://github.com/polylang/polylang-pro/issues/2713}
+	 */
+	public function test_register_custom_table_taxomomy() {
+		add_action(
+			'pll_model_init',
+			function ( $model ) {
+				// Register the DB table in Polylang.
+				$foo = ( new PLLTest_Translatable( $model, 'foo1' ) )->init();
+				$model->translatable_objects->register( $foo );
+			}
+		);
+
+		new PLL_Context_Admin();
+
+		$this->assertSame( array( 'foo1_language' ), get_option( 'pll_language_taxonomies' ) );
+	}
```

Comment:
This seems to have a similar intention as this test https://github.com/polylang/polylang/blob/67d462ae9c3170fddefedc6dc83aa9e0c3a9840b/tests/phpunit/tests/test-model.php#L203 
which unfortunately didn't fail when we introduced the bug. Instead of introducing a new test, the best would be to modify the original test to ensure that it fails on the bug and then add the new assertion.

Reply:
This test doesn't fail because Polylang API is loaded at the beginning.
See https://github.com/polylang/polylang/blob/3.7.3/tests/phpunit/tests/test-model.php#L14
So it can't reproduce what happens really in production even if we trigger the object registration with `do_action( 'pll_model_init' )`.
Indeed `PLL()` is always defined in this case in `Model_Test` test class.
Of course we can add the new assertion to check the new option but we can't here simulate the wrong usage of `PLL()` in the old process.

Done in 941ac51229a05fcc4c013e0f6e1d66755ee225a0

I think the new test is closer of what really happens when the fatal error occurs and should be kept.

Reply:
Only in these two tests 
https://github.com/polylang/polylang/blob/3.7.3/tests/phpunit/tests/test-model.php#L140-L184